app/utils/currencyConversion.ts

```diff
@@ -64,19 +64,33 @@ export const paymentAmountToText = (
   locale = "en-US",
 ): string => {
   if (paymentAmount.currency === WalletCurrency.USD) {
-    return (paymentAmount.amount / 100).toLocaleString(locale, {
-      style: "decimal",
-      maximumFractionDigits: 2,
-      minimumFractionDigits: 2,
-    })
+    try {
+      const amount = paymentAmount.amount / 100
+      if (amount) {
```

Comment:
It seems that this `if` is unnecessary and so is line 76. Since the whole statement is inside of a try block, you can just do the calculation as normal and then catch any errors that are thrown from the function.